pandas/tests/frame/methods/test_join.py

```diff
@@ -104,6 +107,134 @@ def test_suffix_on_list_join():
     tm.assert_frame_equal(arr_joined, norm_joined)
 
 
+def test_join_validation():
```

Comment:
Could you split this into multiple tests so it's easier to see which conditions are being tested?